src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -78,7 +83,7 @@ protected override void Initialize()
                 // start remote host
                 EnableRemoteHostClientService();
 
-                Workspace.AdviseSolutionEvents((IVsSolution)GetService(typeof(SVsSolution)));
+                Workspace.AdviseSolutionEvents((IVsSolution)await GetServiceAsync(typeof(SVsSolution)).ConfigureAwait(true));
```

Comment:
Does GetServiceAsync ensure that the tasks are always completed immediately if the service already exists? My worry here is if CreateWorkspace() _did_ in fact create this, we haven't actually hooked up eventing until later, which means there's a window where we might miss an event. Looking at the code _right now_ it seems safeish, because the only events we're watching right now are solution closed events. A more paranoid change might be to fetch the service earlier and then do the CreateWorkspace() through AdviseSolutionEvents with no await in the middle.

(It also seems odd this is in the language package...does this run twice if we have mixed C#/VB?)

Reply:
> Does GetServiceAsync ensure that the tasks are always completed immediately if the service already exists?

It *probably* behaves that way, but I doubt there's a firm guarantee of that.

Reply:
moving GetServiceAsync up so that there is no yield between workspace and subscribing to event seems safer way and easy thing to do.

are we okay yielding like this while initializing Roslyn? 

does AsyncPackage make sure that until initializeAsync is done, none of service it registers are available to others?

Reply:
:arrow_right: I reordered the service uses in 7eacce858f22ded4d2fc8d65e06450f0d4b57171.